lib/NormalModuleFactory.js

```diff
@@ -104,15 +104,15 @@ const stringifyLoadersAndResource = (loaders, resource) => {
 const identToLoaderRequest = resultString => {
 	const idx = resultString.indexOf("?");
```

Comment:
Rare, but the tricky part here is that `\0?` should not be treated as query string start.

Luckily we already have a helper functions for that: `parseResource` in `utils/identifier.js`

but that also parses the fragment. That should be a breaking changes for cases like: `./loader.js?{"items":["a","b","c#","d"]}`.

So best copy `parseResource` into `parseResourceWithoutFragment` and remove the fragment part, so it can be used there. `parseResourceWithoutFragment` should stay in `identifier.js`

The good thing is that this function is also cached